frontend/src/components/MatchingPairs/MatchingPairs.jsx

```diff
@@ -90,14 +90,18 @@ const MatchingPairs = ({
             if (turnedCards.length === 1) {
                 // This is the second card to be turned
                 currentCard.turned = true;                
-                setSecondCard(currentCard);
+                setSecondCard(currentCard);                                
                 // set no mouse events for all but current
                 sections.forEach(section => section.noevents = true);
                 currentCard.noevents = true;
+                currentCard.boardposition = parseInt(index) + 1;
+                currentCard.timestamp = performance.now();
+                firstCard ? currentCard.response_interval_ms = Math.round(currentCard.timestamp - firstCard.timestamp) : {};
```

Comment:
I've never seen this syntax before: a question mark before assignment? What does this do, is it a conditional assignment? Perhaps add a comment to clarify what the intention is here.

Reply:
Yeah, I think this could be rewritten into something simpler. I'm not entirely sure what the intention is either.

Reply:
I wasn't too happy about this as well, but the test `updates score after a match` kept failing because the value of firstCard was `null`.
I found another solution, I've now set the initial state of firstCard and secondCard to `{}`